nifi-stateless/nifi-stateless-bundle/nifi-stateless-engine/src/main/java/org/apache/nifi/stateless/engine/StatelessFlowManager.java

```diff
@@ -320,9 +321,55 @@ public ReportingTaskNode createReportingTask(final String type, final String id,
 
     @Override
     public ParameterProviderNode createParameterProvider(final String type, final String id, final BundleCoordinate bundleCoordinate, final Set<URL> additionalUrls, final boolean firstTimeAdded,
-                                                         final boolean register) {
+                                                         final boolean registerLogObserver) {
+        if (type == null || id == null || bundleCoordinate == null) {
+            throw new NullPointerException("Must supply type, id, and bundle coordinate in order to create Reporting Task. Provided arguments were type=" + type + ", id=" + id
+                + ", bundle coordinate = " + bundleCoordinate);
+        }
+
+        // make sure the first reference to LogRepository happens outside of a NarCloseable so that we use the framework's ClassLoader
+        final LogRepository logRepository = LogRepositoryFactory.getRepository(id);
+
+        final ParameterProviderNode taskNode;
+        try {
+            taskNode = new ComponentBuilder()
+                    .identifier(id)
+                    .type(type)
+                    .bundleCoordinate(bundleCoordinate)
+                    .statelessEngine(statelessEngine)
+                    .additionalClassPathUrls(additionalUrls)
+                    .flowManager(this)
+                    .buildParameterProvider();
+        } catch (ParameterProviderInstantiationException e) {
+            throw new RuntimeException("Could not create Parameter Context Task of type " + type + " with ID " + id, e);
+        }
 
-        throw new UnsupportedOperationException("Parameter Providers are not supported in Stateless NiFi");
+        LogRepositoryFactory.getRepository(taskNode.getIdentifier()).setLogger(taskNode.getLogger());
+
+        if (firstTimeAdded) {
+            final Class<?> taskClass = taskNode.getParameterProvider().getClass();
+            final String identifier = taskNode.getParameterProvider().getIdentifier();
+
+            try (final NarCloseable x = NarCloseable.withComponentNarLoader(statelessEngine.getExtensionManager(), taskClass, identifier)) {
+                ReflectionUtils.invokeMethodsWithAnnotation(OnAdded.class, taskNode.getParameterProvider());
+
+                if (isFlowInitialized()) {
+                    ReflectionUtils.quietlyInvokeMethodsWithAnnotation(OnConfigurationRestored.class, taskNode.getParameterProvider(), taskNode.getConfigurationContext());
+                }
+            } catch (final Exception e) {
+                throw new ComponentLifeCycleException("Failed to invoke On-Added Lifecycle methods of " + taskNode.getParameterProvider(), e);
+            }
+        }
+
+        if (registerLogObserver) {
+            onParameterProviderAdded(taskNode);
+
+            // Register log observer to provide bulletins when reporting task logs anything at WARN level or above
```

Comment:
This comment appears to be copied from Reporting Tasks, so it should be updated or removed.